artifactory/commands/transferfiles/manager.go

```diff
@@ -182,19 +190,39 @@ func (ptm *PollingTasksManager) stop() {
 }
 
 type producerConsumerWrapper struct {
-	chunkBuilderProducerConsumer  parallel.Runner
 	chunkUploaderProducerConsumer parallel.Runner
+	chunkBuilderProducerConsumer  parallel.Runner
+	uploaderFinishedNotifier      chan bool
+	builderFinishedNotifier       chan bool
 	errorsQueue                   *clientUtils.ErrorsQueue
 }
 
+// Notifies that the uploader finished if there are no builders and the current thread is the last one
```

Comment:
Notifies that the uploader finished if there are no builders and the current thread is the last one
-->
Sends a signal through the `uploaderFinishedNotifier` channel, if the `chunkUploaderProducerConsumer` finished running all of its tasks.

Let's also explain as part of the above description the dependency between the two producer-consumers, and how this dependency affects the decision of whether the uploader finished.

Reply:
done